src/librbd/image/RemoveRequest.cc

```diff
@@ -69,6 +69,8 @@ Context *RemoveRequest<I>::handle_open_image(int *result) {
   if (*result < 0) {
     lderr(m_cct) << "error opening image: " << cpp_strerror(*result) << dendl;
     m_retval = *result;
+    if (*result != -ENOENT)
+      m_retval = *result;
```

Comment:
yes, should not add it here. will remove it at first, then I will focus on how to fix it properly in #12638 . thanx for your suggestion.